src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = newNode;
+        } else {
+            tail.setNext(newNode);
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdd(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> current = getNode(index);
+            Node<T> newNode = new Node<>(current.getPrev(), value, current);
+            if (current.getPrev() == null) {
+                head = newNode;
+            } else {
+                current.getPrev().setNext(newNode);
```

Comment:
Direct field access should be used instead of setters for the inner class `Node`. Since `Node` should be an inner class, you can access its fields directly if they are declared private.